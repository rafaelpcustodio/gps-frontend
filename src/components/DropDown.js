import PropTypes from 'prop-types'
import React from 'react'

import { colors } from '../helpers/colors'

import Icon from './Icon'
import If from './If'
import StyledDropdown from './StyledDropdown'

import { faCaretDown, faCaretUp } from '@fortawesome/free-solid-svg-icons'

const Dropdown = props => {

    const { 
        dark, 
        disabled, 
        expanded 
    } = props

    const iconColor = disabled ? colors.disabled : colors.text
    
    return (
        <StyledDropdown
            className={ expanded ? 'is-expanded' : null }
            dark={ dark }
            expanded={ expanded }>
            {  
                <If condition={ expanded } el={ 
                    <Icon color={ iconColor } icon={faCaretDown} size={"lg"} /> 
                    }>
                    <Icon color={ iconColor } icon={faCaretUp} size={"lg"} />
                </If>
            }
        </StyledDropdown>
    )
}

Dropdown.defaultProps = {
    disabled: false,
    expanded: false
}

Dropdown.propTypes = {
    dark: PropTypes.bool,
    disabled: PropTypes.bool.isRequired,
    expanded: PropTypes.bool.isRequired
}

export default Dropdown
