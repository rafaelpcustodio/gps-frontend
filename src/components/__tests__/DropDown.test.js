import configureMockStore from 'redux-mock-store'
import { Map, Record } from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import { Provider } from 'react-redux'

import DropDown from '../DropDown'

describe('Tests for DrowDown component', () => {
    let store = {}
    let mockStore = {}

    it('Should render without crash', () => {
        mockStore = configureMockStore()
        store = mockStore(Record({
            gpsReducers: Map({
            })
        })())

        const tree = renderer.create(
            <Provider store={ store }>
                <DropDown
                    dark={true}
                    disabled={false} 
                    expanded={false}
                />
            </Provider>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})
