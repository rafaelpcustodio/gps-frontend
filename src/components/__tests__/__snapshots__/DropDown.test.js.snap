// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for DrowDown component Should render without crash 1`] = `
<button
  className="StyledDropdown-sc-1d17mlt-0 crWdhk"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-caret-down fa-w-10 fa-lg "
    color="#363636"
    data-cy="icon"
    data-icon="caret-down"
    data-prefix="fas"
    focusable="false"
    role="img"
    style={Object {}}
    viewBox="0 0 320 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</button>
`;
